server/ip-tools.ts

```diff
@@ -354,158 +356,128 @@ export const IPTools = new class {
+		) {
+			return 'ideacellular.mobile-nohost';
+		}
+		if (IPTools.checkPattern(rangeTmobile, ipNumber) || ip.startsWith('149.254.')) {
+			return 'tmobile.mobile-nohost';
+		}
+		if (
+			IPTools.checkPattern(rangeCenet, ipNumber) || IPTools.checkPattern(rangeQlded, ipNumber) ||
+			ip.startsWith('153.107.') || IPTools.checkPattern(rangeCathednet, ipNumber)
+		) {
+			return 'edu.au.res-nohost';
+		}
+		if (ip.startsWith('179.7.')) {
+			return 'claro.com.pe.mobile-nohost';
+		}
+		if (ip.startsWith('190.') || IPTools.checkPattern(rangeTelefonica, ipNumber)) {
+			return 'telefonica.net.pe.mobile-nohost';
+		}
+		if (ip.startsWith('180.191.') || ip.startsWith('112.198.')) {
+			return 'globe.com.ph.mobile-nohost';
+		}
+		if (ip.startsWith('218.188.') || ip.startsWith('218.189.')) {
+			return 'hgc.com.hk.mobile-nohost';
+		}
+		if (ip.startsWith('172.242.') || ip.startsWith('172.243.')) {
+			return 'viasat.com.mobile-nohost';
+		}
+		if (ip.startsWith('201.141.')) {
+			return 'cablevision.net.mx.mobile-nohost';
+		}
+		if (IPTools.checkPattern(rangeStarhub, ipNumber)) {
+			return 'starhub.com.mobile-nohost';
+		}
+		if (ip.startsWith('202.12.94.') || ip.startsWith('202.12.95.')) {
+			return 'nyp.edu.sg.res-nohost';
+		}
+		if (ip.startsWith('64.150.')) {
+			return 'illinois.net.res-nohost';
+		}
+		if (ip.startsWith('147.129.')) {
+			return 'ithaca.edu.res-nohost';
+		}
+		if (ip.startsWith('189.204.')) {
+			return 'bestel.com.mx.res-nohost';
+		}
+		if (IPTools.checkPattern(rangePsci, ipNumber)) {
+			return 'psci.net.res-nohost';
+		}
+		if (IPTools.checkPattern(rangeOcde, ipNumber)) {
+			return 'ocde.us.res-nohost';
+		}
+		if (IPTools.checkPattern(rangeIhet, ipNumber)) {
+			return 'iu.edu.res-nohost';
+		}
+		if (IPTools.checkPattern(rangeTimcelular, ipNumber)) {
+			return 'tim.com.br.mobile-nohost';
+		}
+		if (ip.startsWith('121.54.')) {
+			return 'smart.com.ph.mobile-nohost';
+		}
+		if (ip.startsWith('179.52.') || ip.startsWith('179.53.')) {
+			return 'codetel.net.do.mobile-nohost';
+		}
+		if (ip.startsWith('46.16.36.')) {
+			return 'anchorfree.proxy-nohost';
+		}
+		if (
+			ip.startsWith('198.144.104.') || ip.startsWith('198.47.115.') || ip.startsWith('199.255.215.') ||
+			ip.startsWith('204.14.76.') || ip.startsWith('204.14.77.') || ip.startsWith('204.14.78.') ||
+			ip.startsWith('204.14.79.') || ip.startsWith('205.164.32.') || ip.startsWith('209.73.132.') ||
+			ip.startsWith('209.73.151.') || ip.startsWith('216.172.135.') || ip.startsWith('46.16.34.') ||
+			ip.startsWith('46.16.35.') || ip.startsWith('50.117.45.') || ip.startsWith('63.141.198.') ||
+			ip.startsWith('63.141.199.') || ip.startsWith('74.115.1.') || ip.startsWith('74.115.5.') ||
+			ip.startsWith('85.237.197.') || ip.startsWith('85.237.222.')
+		) {
+			return 'anchorfree.proxy-nohost';
+		}
+		if (ip === '127.0.0.1') {
+			return 'localhost';
+		}
+		let hosts;
+		try {
+			hosts = await util.promisify(dns.reverse)(ip);
```

Comment:
I realllllllllly hate `util.promisify`...

It should at least not be used like this.

Reply:
```suggestion
			hosts = new Promise(resolve => {
				dns.reverse(ip, (err, hosts) => resolve(hosts));
			});
```